sycl/doc/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,493 @@
+# SYCL-Graph Usage Guide and Examples
+
+This document describes recommended usage guidelines for using the
+`sycl_ext_oneapi_graph` extension (referred to as SYCL-Graph in this document)
+as well as provides various example code snippets for various features and usage
+scenarios.
+
+The specification for the `sycl_ext_oneapi_graph` extension can be found
+[here](./extensions/experimental/sycl_ext_oneapi_graph.asciidoc).
+
+## General Usage Guidelines
+
+The following section provides some general usage guidelines when working
+with SYCL-Graph or adapting an existing SYCL application to use SYCL-Graph.
+Examples here will generally only use one graph creation API ("Explicit" or
+"Record & Replay") for simplicity, but can generally be applied to either API
+unless specifically noted.
+
+### Use Host-Tasks For Host Work
+
+SYCL-Graph cannot capture work done on the host in application code which does
+not go through the SYCL runtime. Any work to be done on the host should be
```

Comment:
Here it might be also worth to mention that direct kernel submissions to lower-level API, such as Level Zero or CUDA are not captured ...